packages/shorebird_cli/lib/src/artifact_builder.dart

```diff
@@ -64,6 +65,9 @@ class ArtifactBuilder {
         executable,
         arguments,
         runInShell: true,
+        environment: base64PublicKey != null
+            ? {'SHOREBIRD_PUBLIC_KEY': base64PublicKey}
```

Comment:
Since we're ultimately setting an environment variable couldn't this PR be simplified and check for the presence of this environment variable instead of adding a command-line option?

Reply:
It could, but having the user provide a path to a file as a CLI arg is more in-line with how our system currently operates. I think we'd ideally pass this to the flutter tool as an arg as well, but I believe it was easier to do via an env var (@erickzanardo could speak more on this part)

Reply:
Yeah, this is done this way since passing an argument to the code that bundles the shorebird.yaml file on android is not trivial.

That is done in the gradle layer (https://github.com/shorebirdtech/flutter/blob/shorebird/dev/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy#L1337), and AFAIK, to pass an argument there, we would need to do multiple changes in the gradle tasks, I might be wrong on this since gradle is something I am still learning, but that is how we managed to work so far.

Happy to revisit this eventually, but I feel that in favor of shipping code signing soon, I guess this is good enough.

Let me know if any of you thinks otherwise

Reply:
Sounds good I only asked because it seemed like a lot more work to modify release commands to accept a command-line arg just to pass an environment variable to the flutter tool. Seemed much simpler to just provide the public key as an environment variable from the start since folks will likely be doing that anyway in CI environment via GitHub secrets manager etc.